azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py

```diff
@@ -309,6 +368,29 @@ def _encode_properties(self, message_to_send, topic):
 
         return topic
 
+    def _extract_properties(self, properties, message_received):
+        key_value_pairs = properties.split("&")
+
+        for entry in key_value_pairs:
+            pair = entry.split("=")
+            key = urllib.parse.unquote_plus(pair[0])
+            value = urllib.parse.unquote_plus(pair[1])
+
+            if key == "$.mid":
+                message_received.message_id = value
+            elif key == "$.cid":
+                message_received.correlation_id = value
+            elif key == "$.uid":
+                message_received.user_id = value
+            elif key == "$.to":
+                message_received.to = value
+            elif key == "$.ct":
```

Comment:
It feels like this should be table driven, along with the matching code up in _encode_properties above.

Reply:
i am not sure what this means...does it mean creating a dictionary whether keys are constants and values are "$.mid" and "$.cid"

---
In reply to: [246548387](https://github.com/Azure/azure-iot-sdk-python-preview/pull/36#discussion_r246548387) [](ancestors = 246548387)